azurerm/internal/services/bot/resource_arm_bot_channel_directline.go

```diff
@@ -0,0 +1,341 @@
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Channel Directline for Bot %q (Resource Group %q): %+v", id.ResourceGroup, botName, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", resp.Location)
+	d.Set("bot_name", botName)
+
+	channelsResp, _ := client.ListWithKeys(ctx, id.ResourceGroup, botName, botservice.ChannelNameDirectLineChannel)
+	if props := channelsResp.Properties; props != nil {
+		if channel, ok := props.AsDirectLineChannel(); ok {
+			if channelProps := channel.Properties; channelProps != nil {
+				d.Set("site", unExpandSites(*channelProps.Sites))
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmBotChannelDirectlineUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Bot.ChannelClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	botName := d.Get("bot_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	sites, _ := expandSites(d.Get("site").(*schema.Set).List())
+
+	channel := botservice.BotChannel{
+		Properties: botservice.DirectLineChannel{
+			Properties: &botservice.DirectLineChannelProperties{
+				Sites: sites,
+			},
+			ChannelName: botservice.ChannelNameDirectLineChannel1,
+		},
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Kind:     botservice.KindBot,
+	}
+
+	if _, err := client.Update(ctx, resourceGroup, botName, botservice.ChannelNameDirectLineChannel, channel); err != nil {
+		return fmt.Errorf("Error issuing create request for Channel Directline for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameDirectLineChannel1))
+	if err != nil {
+		return fmt.Errorf("Error making get request for Channel Directline for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Channel Directline for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBotChannelDirectlineRead(d, meta)
+}
+
+func resourceArmBotChannelDirectlineDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Bot.ChannelClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	botName := id.Path["botServices"]
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, botName, string(botservice.ChannelNameDirectLineChannel1))
+	if err != nil {
+		if !response.WasNotFound(resp.Response) {
+			return fmt.Errorf("Error deleting Channel Directline for Bot %q (Resource Group %q): %+v", id.ResourceGroup, botName, err)
+		}
+	}
+
+	return nil
+}
+
+func expandSites(input []interface{}) (*[]botservice.DirectLineSite, error) {
```

Comment:
we are note returning an error anywhere i can see so
```suggestion
func expandSites(input []interface{}) *[]botservice.DirectLineSite {
```